### PR TITLE
fix(controller): Adds PNS_PRIVILEGED, fixed termination bug

### PR DESCRIPTION
**Pull request by alexec**
I've removed the need to run ALL tests. 

New suites:

* ArtifactsSuite - tests around artifacts
* SignalsSuite - test related to signals sent to executor (e.g. SIGTEM)

I've deleted a few tests (see my comments).

Fixed a bug that marked some workflows as  an error when they should have been failed.

Added PNS_PRIVILEGED to run PNS in privileged mode, sometimes needed for GCP.

### CODE REVIEW

File: test/e2e/functional_test.go [hunk 1]
```diff
@@ -293,25 +281,6 @@ spec:
 		})
 }
 
-func (s *FunctionalSuite) TestFastFailOnPodTermination() {
```
Comment (alexec) [inline test/e2e/functional_test.go:296]: deleted as skipped

File: test/e2e/functional_test.go [hunk 2]
```diff
@@ -554,87 +523,6 @@ func (s *FunctionalSuite) TestParameterAggregation() {
 		})
 }
 
-func (s *FunctionalSuite) TestGlobalScope() {
-	s.Need(fixtures.BaseLayerArtifacts)
```
Comment (alexec) [inline test/e2e/functional_test.go:558]: deleted - this test is flakey and it was a time suck fixing it

File: workflow/controller/operator.go [hunk 3]
```diff
@@ -1116,38 +1116,29 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 		}
 		newDaemonStatus = pointer.BoolPtr(false)
 	case apiv1.PodRunning:
-		if pod.DeletionTimestamp != nil {
```
Comment (alexec) [inline workflow/controller/operator.go:1119]: this condition is incorrect - a pod with a deletion timestamp is one we may have caused to be deleted outselver

**Review by simster7 — APPROVED** [review]

**Comment by lnguyen** [comment]
can this be cherry picked for v2.x?

**Comment by alexec** [comment]
This is too big to back-port all of it really. Is there a specific thing you need from it?

**Comment by lnguyen** [comment]
will it be easy transition to v3.x? no specific need, just running this using onepanel on microk8s with pns and getting alot of crashes

**Comment by alexec** [comment]
Most user should not have any migration issues to v3.0

**Pull request merged**